storage/src/mmr/mutable.rs

```diff
@@ -237,212 +323,341 @@ impl<K: Array, V: Array, H: CHasher> MutableMmr<K, V, H> {
-            pinned_nodes,
+    /// Sync the store to disk ensuring the current state is persisted.
+    async fn sync(&mut self) -> Result<(), Error> {
+        futures::try_join!(
+            self.log.sync().map_err(Error::JournalError),
+            self.ops.sync()
+        )?;
+
+        Ok(())
+    }
+
+    /// Close the store. Operations that have not been committed will be lost.
+    pub async fn close(self) -> Result<(), Error> {
+        if self.uncommitted_ops > 0 {
+            warn!(
+                op_count = self.uncommitted_ops,
+                "closing store with uncommitted operations"
+            );
         }
+        futures::try_join!(
+            self.log.close().map_err(Error::JournalError),
+            self.ops.close()
+        )?;
+
+        Ok(())
     }
 
     /// Raise the inactivity floor by exactly `max_steps` steps. Each step either advances over an
     /// inactive operation, or re-applies an active operation to the tip and then advances over it.
     ///
     /// This method does not change the state of the store's snapshot, but it always changes the
     /// root since it applies at least one (floor) operation.
-    #[allow(dead_code)] // TODO: Remove when method gets used.
-    pub fn raise_inactivity_floor(&mut self, hasher: &mut H, max_steps: u64) {
+    async fn raise_inactivity_floor(
+        &mut self,
+        hasher: &mut H,
+        max_steps: u64,
+    ) -> Result<(), Error> {
         for _ in 0..max_steps {
-            let i = self.loc_to_op_index(self.inactivity_floor_loc);
-            if i == self.log.len() {
+            if self.inactivity_floor_loc == self.op_count() {
                 break;
             }
-            let op = &self.log[i];
+
+            let op = self.log.read(self.inactivity_floor_loc).await?;
             let op_count = self.op_count();
             let key = op.to_key();
             if let Some(loc) = self.snapshot.get_mut(&key) {
                 if *loc == self.inactivity_floor_loc {
                     // This operation is active, move it to tip to allow us to continue raising the
                     // inactivity floor.
                     *loc = op_count;
-                    self.apply_op(hasher, op.clone());
+                    self.apply_op(hasher, op.clone()).await?;
                 }
             }
             self.inactivity_floor_loc += 1;
         }
-        self.apply_op(hasher, Operation::floor(self.inactivity_floor_loc));
+        self.apply_op(hasher, Operation::floor(self.inactivity_floor_loc))
+            .await?;
+
+        Ok(())
     }
 
     /// Prune any historical operations that are known to be inactive (those preceding the
     /// inactivity floor). This does not affect the store's root or current snapshot.
-    pub fn prune_known_inactive(&mut self) {
-        let pruned_ops = self.inactivity_floor_loc - self.pruned_loc;
+    pub async fn prune_known_inactive(&mut self) -> Result<(), Error> {
```

Comment:
How do you feel about calling this at the end of commit (now that the flow is much more rigid, I think it no longer makes sense for this to be `public`).

Reply:
as discussed, yes, good plan. done

Reply:
...although could there be use cases where someone would want to retain the update history?   (I know you're not a fan of adding any complexity for "archival" state.)

Reply:
Interesting point. Thinking 🤔 .

Reply:
Let's change the name of `proof_to_tip` to just be `proof` btw.

Reply:
done